tests/encoding_test.c

```diff
@@ -997,3 +997,121 @@ static int s_hex_encoding_append_dynamic_test_case_empty(struct aws_allocator *a
 }
 
 AWS_TEST_CASE(hex_encoding_append_dynamic_test_case_empty, s_hex_encoding_append_dynamic_test_case_empty)
+
+static int read_file_contents(struct aws_byte_buf *out_buf, struct aws_allocator *alloc, const char *filename) {
+    AWS_ZERO_STRUCT(*out_buf);
+    FILE *fp = fopen(filename, "r");
+
+    if (fp) {
+        if (fseek(fp, 0L, SEEK_END)) {
+            fclose(fp);
+            return AWS_OP_ERR;
+        }
+
+        size_t allocation_size = (size_t)ftell(fp) + 1;
+        /* Tell the user that we allocate here and if success they're responsible for the free. */
+        if (aws_byte_buf_init(out_buf, alloc, allocation_size)) {
+            fclose(fp);
+            return AWS_OP_ERR;
+        }
+
+        /* Ensure compatibility with null-terminated APIs, but don't consider
+         * the null terminator part of the length of the payload */
+        out_buf->len = out_buf->capacity - 1;
+        out_buf->buffer[out_buf->len] = 0;
+
+        if (fseek(fp, 0L, SEEK_SET)) {
+            aws_byte_buf_clean_up(out_buf);
+            fclose(fp);
+            return AWS_OP_ERR;
+        }
+
+        size_t read = fread(out_buf->buffer, 1, out_buf->len, fp);
+        fclose(fp);
+        if (read < out_buf->len) {
+            aws_secure_zero(out_buf->buffer, out_buf->len);
```

Comment:
there's a secure cleanup of byte buf

Reply:
That's copy pasta from aws_byte_buf_init_from_file(), I'll just remove it, it doesn't need to be secure.